docs/content/blog/authelia-traefik-setup-guide/index.md

```diff
@@ -0,0 +1,374 @@
+## Assumptions and Adaptation
+This guide makes a few assumptions. These assumptions may require adaptation in more advanced and complex scenarios. We can not reasonablly have examples for every advanced configuration option that exists. Some of these values can be automatically replaced with documentation variables.
+
+{{< sitevar-preferences >}}
+
+We make the following assumptions:
+- [Docker](https://docs.docker.com/engine/install/) is configured correctly.
+- Single Host
+- You will have to adapt all instances of `{{< sitevar name="host" nojs="authelia" >}}` in the URL if:
+  - you're using a different container name
+  - you deployed the proxy to a different location
+- You will have to adapt all instances of `{{< sitevar name="port" nojs="9091" >}}` in the URL if:
+  - you have adjusted the default port in the configuration
+- You will have to adapt the entire URL if:
+  - Authelia is on a different host to the proxy
+- All services are part of the `{{< sitevar name="domain" nojs="example.com" >}}` domain:
+- This domain and the subdomains will have to be adapted in all examples to match your specific domains unless you're
+  just testing, or you want to use that specific domain
+
+## File Structure
+The first thing we want to do is set up the file structure. Which should look something like this:
+```text
+📁 project
+ ┣ 📁 authelia
+ ┃  ┣ 📁 config
+ ┃  ┃ ┣ 📄 configuration.yaml
+ ┃  ┃ ┗ 📄 users.yaml
+ ┃  ┗ 📁 secrets
+ ┣ 📄 compose.yaml
+ ┗ 📁 traefik
+    ┣ 📁 config
+    ┃ ┣ 📄 dynamic.yaml
+    ┃ ┗ 📄 traefik.yaml
+    ┣ 📁 data
+    ┃ ┗ 📄 acme.json
+    ┣ 📁 logs
+    ┗ 📁 secrets
+```
+
+## Traefik and Whoami
+{{< callout context="note" title="Note" icon="outline/info-circle" >}}
+We'll focus on the minimal configuration needed to work with Authelia. For advanced Traefik features and configurations, consult their documentation.
+{{< /callout >}}
+
+Next, we'll set up Traefik as our reverse proxy. For detailed Traefik documentation, refer to the [official Traefik docs](https://doc.traefik.io/traefik/).
+
+#### Docker Compose
+
+
+```yaml{title="compose.yaml"}
+services:
+  traefik:
+    image: traefik:latest
+    container_name: traefik
+    restart: unless-stopped
+    security_opt:
+      - no-new-privileges=true
+    networks:
+      proxy: {}
+      authelia:
+        aliases:
+          - '{{< sitevar name="subdomain-authelia" nojs="auth" >}}.{{< sitevar name="domain" nojs="example.com" >}}'
+    ports:
+      - '80:80'
+      - '443:443'
+    environment:
+      TZ: 'America/Los_Angeles' # see below
+      CF_EMAIL_FILE: '/run/secrets/cloudflare_email'
+      CF_DNS_API_TOKEN_FILE: '/run/secrets/cloudflare_api_key'
+    volumes:
+      - '/var/run/docker.sock:/var/run/docker.sock:ro'
+      - './traefik/config/traefik.yaml:/traefik.yaml:ro'
+      - './traefik/config/dynamic.yaml:/dynamic.yaml:ro'
+      - './traefik/data/:/data'
+      - './traefik/logs:/logs'
+    secrets:
+      - cloudflare_email
+      - cloudflare_api_key
```

Comment:
These secrets are not being mentioned in the secrets section

Reply:
We should probably avoid using anything platform specific like this and instead use HTTP-01 ACME challenges.